### PR TITLE
feature: allow installing source trees

### PR DESCRIPTION
**Pull request by rgrinberg**
This PR allows us to install source trees with the following field in the install stanza:

```
(install
 (section doc)
 (source_tree dir))
```

### CODE REVIEW

File: src/dune_rules/install_rules.ml [hunk 1]
```diff
@@ -1033,15 +1083,20 @@ include (
...
-          |> Install.Entry.gen_install_file
+        let open Fiber.O in
+        let+ entries =
+          let+ entries =
+            Fiber.parallel_map entries ~f:(fun (entry : _ Install.Entry.t) ->
+              match entry.kind with
+              | `File -> Fiber.return [ entry ]
+              | `Directory -> Fiber.return (read_dir_recursively entry)
+              | `Source_tree ->
+                (* these are expanded into [`File] *)
```
Comment (gridbugs) [inline src/dune_rules/install_rules.ml:1094]: nitpick: the `{` should be a `[`

File: src/dune_rules/install_rules.ml [hunk 2]
```diff
@@ -1033,15 +1083,20 @@ include (
...
+                assert false)
```
Comment (gridbugs) [inline src/dune_rules/install_rules.ml:1095]: Should this be a `Code_error`?

**Review by gridbugs — APPROVED** [review]

File: src/dune_rules/install_rules.ml (hunk 1 above)
Comment (rgrinberg) [reply src/dune_rules/install_rules.ml:1094]: Did you attach your comment to the wrong line? There's no `{` on this line`.

Comment (gridbugs) [reply src/dune_rules/install_rules.ml]: ah you're right, for some reason I saw the `[` as a `{`

**Pull request merged**